syncstorage/src/web/middleware/sentry.rs

```diff
@@ -161,12 +169,15 @@ where
 /// `sentry::event_from_error` can't access `std::Error` backtraces as its
 /// `backtrace()` method is currently Rust nightly only. This function works
 /// against `HandlerError` instead to access its backtrace.
-pub fn event_from_error(err: &ApiError) -> Event<'static> {
+pub fn event_from_error<E>(err: &E) -> Event<'static>
+where
+    E: ErrorBacktrace + StdError + 'static,
+{
     let mut exceptions = vec![exception_from_error_with_backtrace(err)];
 
     let mut source = err.source();
     while let Some(err) = source {
-        let exception = if let Some(err) = err.downcast_ref() {
+        let exception = if let Some(err) = err.downcast_ref::<E>() {
```

Comment:
If I understand correctly, this is drilling down to the original source of the error by repeatedly calling `Error::source`. The type parameter was previously omitted, but before this change, I _believe_ that `err.downcast_ref()` was being inferred to be `err.downcast_ref::<ApiError>()`.

I just want to make sure that explicitly passing `E` as the type param here doesn't change the functionality at all

Reply:
That's right, so now it's `Reportable + etc` so it should work the same AFAICT.